cvat/apps/engine/static/engine/js/shapeCollection.js

```diff
@@ -221,49 +229,84 @@ class ShapeCollectionModel extends Listener {
 
-    export() {
-        let response = {
-            "boxes": [],
-            "box_paths": [],
-            "points": [],
-            "points_paths": [],
-            "polygons": [],
-            "polygon_paths": [],
-            "polylines": [],
-            "polyline_paths": [],
-        };
+        if (export_type === ExportType.create) {
+            export_action_container = container.create;
+        } else if (export_type === ExportType.update) {
+            export_action_container = container.update;
+        } else if (export_type === ExportType.delete) {
+            export_action_container = container.delete;
+        }
 
-        for (let shape of this._shapes) {
-            if (shape.removed) continue;
-            switch (shape.type) {
-            case 'annotation_box':
-                response.boxes.push(shape.export());
-                break;
-            case 'interpolation_box':
-                response.box_paths.push(shape.export());
-                break;
-            case 'annotation_points':
-                response.points.push(shape.export());
-                break;
-            case 'interpolation_points':
-                response.points_paths.push(shape.export());
-                break;
-            case 'annotation_polygon':
-                response.polygons.push(shape.export());
-                break;
-            case 'interpolation_polygon':
-                response.polygon_paths.push(shape.export());
-                break;
-            case 'annotation_polyline':
-                response.polylines.push(shape.export());
-                break;
-            case 'interpolation_polyline':
-                response.polyline_paths.push(shape.export());
-            }
+        if (!export_action_container) {
+            throw Error('Undefined action of shape');
         }
 
-        return JSON.stringify(response);
+        switch (shape_type) {
+        case 'annotation_box':
+            shape_container_target = export_action_container.boxes;
+            break;
+        case 'interpolation_box':
+            shape_container_target = export_action_container.box_paths;
+            break;
+        case 'annotation_points':
+            shape_container_target = export_action_container.points;
+            break;
+        case 'interpolation_points':
+            shape_container_target = export_action_container.points_paths;
+            break;
+        case 'annotation_polygon':
+            shape_container_target = export_action_container.polygons;
+            break;
+        case 'interpolation_polygon':
+            shape_container_target = export_action_container.polygon_paths;
+            break;
+        case 'annotation_polyline':
+            shape_container_target = export_action_container.polylines;
+            break;
+        case 'interpolation_polyline':
+            shape_container_target = export_action_container.polyline_paths;
+        }
+
+        if (!shape_container_target) {
```

Comment:
Does switch has 'default'? Does it make sense to move the statement into default?